csharp/src/Apache.Arrow.Flight/Client/FlightClient.cs

```diff
@@ -93,6 +93,22 @@ public FlightRecordBatchDuplexStreamingCall StartPut(FlightDescriptor flightDesc
                 channels.Dispose);
         }
 
+        public AsyncDuplexStreamingCall<FlightHandshakeRequest, FlightHandshakeResponse> Handshake(Metadata headers = null)
```

Comment:
From what I can tell most of the GRPC underlying objects are constructed this way in the `FlightGrpc` generated output.  There is a similar call for `DoPut` where the response is also an `AsyncDuplexStreamingCall`  but the method is synchronous as well.  The handshake can be a multi step process based on the documentation, that is my best guess as to why it is structured this way.  

When I tested with the Java and Python clients, they would not connect without the handshake method being implemented.  Otherwise I would have left this out of the PR.